apps/web/app/modules/Admin/EditCourse/CourseLessons/components/LessonCard.tsx

```diff
@@ -0,0 +1,67 @@
+import { useMemo } from "react";
+
+import { Icon } from "~/components/Icon";
+
+import { mapItemType, mapTypeToIcon } from "../CourseLessons.helpers";
+
+import type { LessonItem } from "../../EditCourse.types";
+import type { IconName } from "~/types/shared";
+
+interface LessonCardProps {
+  item: LessonItem;
+  onClickLessonCard: (lesson: LessonItem) => void;
+}
+
+const LessonCard = ({ item, onClickLessonCard }: LessonCardProps) => {
+  const contentType = item.lessonItemType === "file" ? item.content.type : item.lessonItemType;
+
+  const mappedItemType = useMemo(() => mapItemType(contentType as string), [contentType]);
```

Comment:
I might be the ts-pattern fanboy but if you make the `mapItemType()` like this 
```ts
export const mapItemType = (itemType: string | undefined): string =>
  match(itemType)
    .with("text_block", () => "Text")
    .with("video", () => "Video")
    .with("presentation", () => "Presentation")
    .with("question", () => "Quiz")
    .otherwise(() => "Quiz");
```
then you dont need the type casting because undefined is handled inside :D